### DISCUSSION TITLE
[Bug]: Cannot destructure property 'el' of 'this.lower' as it is undefined

### DISCUSSION THREAD

**Discussion by vaynevayne** [fabricjs/fabric.js#10321] [General]
![image](https://github.com/user-attachments/assets/046d1c1c-aea7-472b-a613-381815788a0e)

What is `this.lower`?
I received this error report more than once.
Since 4.5

**Comment by asturur** [comment]
Since 4.5 you mean 6.4.5?

That may happen when you are working on a disposed canvas.
this.lower is an object containing 2 properties el and ctx that are the element and the context of the canvas you are drawing on

**Reply by vaynevayne** [reply]
Yes, I use 6.4.5. When will the canvas be abandoned? I didn't uninstall the operation

**Reply by asturur** [reply]
It can happen with react on development mode that runs the hook twice for example, or it can happen if you have async operations when your canvas gets updated.

That get()?.canvas reference may be of an old instance that has been wiped by react.
There isn't much context to understand

**Reply by vaynevayne** [reply]
thanks

**Reply by vikhyatsingh123** [reply]
@asturur is there any fix for this? We're getting this error multiple times in our production.
What happens is that when the user tries to move to one screen from another screen, we do canvas.dispose() in the useEffect return and initialize the canvas in the opened screen. But, there maybe chance that somewhere we called canvas.requestRenderAll() which will call on next animation frame. And, by that time, canvas will not present as we already disposed it and we got this error in the console log.
